client/app/visualizations/VisualizationRenderer.jsx

```diff
@@ -0,0 +1,89 @@
+import { isArray } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+import { react2angular } from 'react2angular';
+import { Filters, FiltersType, filterData } from '@/components/Filters';
+import { createPromiseHandler } from '@/lib/utils';
+import { registeredVisualizations, VisualizationType } from './index';
+
+function chooseFilters(globalFilters, localFilters) {
+  return isArray(globalFilters) && (globalFilters.length > 0) ? globalFilters : localFilters;
+}
+
+export class VisualizationRenderer extends React.Component {
+  static propTypes = {
+    visualization: VisualizationType.isRequired,
+    queryResult: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+    filters: FiltersType,
+  };
+
+  static defaultProps = {
+    filters: [],
+  };
+
+  state = {
+    allRows: [], // eslint-disable-line
+    data: { columns: [], rows: [] },
+    filters: this.props.filters, // use global filters by default, if available
+  };
+
+  handleQueryResult = createPromiseHandler(
+    queryResult => queryResult.toPromise(),
+    () => {
+      const { queryResult } = this.props;
+      const columns = queryResult ? queryResult.getColumns() : [];
+      const rows = queryResult ? queryResult.getData() : [];
+      this.setState({
+        allRows: rows, // eslint-disable-line
+        data: { columns, rows },
+      });
+      this.applyFilters(
+        // If global filters available, use them, otherwise get new local filters from query
+        chooseFilters(this.props.filters, queryResult.getFilters()),
```

Comment:
Tiny nit - can swap for:
```js
const { queryResult, filter } = this.props;
...
this.applyFilters(
  // If global filters available, use them, otherwise get new local filters from query
  !isEmpty(filters) ? filters : queryResult.getFilters()
);
```
Cause `!isEmpty` is shorter, and it won't run `queryResult.getFilters()` needlessly.

Reply:
@ranbena Refresh PR - I changed that code this morning because it didn't work as in `master` :sweat_smile: